fs2-stream/src/main/scala/neotypes/fs2/implicits.scala

```diff
@@ -9,7 +9,7 @@ object implicits {
         fs2.Stream.repeatEval(F.suspend(value())).unNoneTerminate
 
       override def onComplete[T](s: fs2.Stream[F, T])(f: => F[Unit]): fs2.Stream[F, T] =
-        s ++ fs2.Stream.eval_(f)
+        s.onFinalize(f)
```

Comment:
The previous implementation was not correct. Since, in case of any error, the `f` function _(namely the rollback)_ would not be executed.

In contrast, since `onFinalize` comes from **Bracket**, it is guarantee that [_"the effectful action is executed at the end of this stream, regardless of how the stream terminates"_](https://oss.sonatype.org/service/local/repositories/releases/archive/co/fs2/fs2-core_2.12/1.0.1/fs2-core_2.12-1.0.1-javadoc.jar/!/fs2/Stream.html#onFinalize[F2[x]%3E:F[x]](f:F2[Unit])(implicitF2:cats.Applicative[F2]):fs2.Stream[F2,O]).